replica/replica.go

```diff
@@ -776,7 +778,13 @@ func (r *Replica) createNewHead(oldHead, parent, created string) (types.DiffDisk
 	}
 
 	if _, err := os.Stat(r.diskPath(newHeadName)); err == nil {
-		return nil, disk{}, fmt.Errorf("%s already exists", newHeadName)
+		logrus.Warningf("Head file: %v already exists", newHeadName)
+		if r.getDiskSize(newHeadName) > 0 {
+			return nil, disk{}, fmt.Errorf("Can't remove head file %v as it contains some data", newHeadName)
+		}
+		if err = r.rmDisk(newHeadName); err != nil {
+			logrus.Errorf("Fail to remove disk, err: %v", err)
```

Comment:
lets return error in case of rmDisk also which is supposed to delete the file successfully

Reply:
and change log message as: Failed to remove already existing disk

Reply:
In rmDisk, errNotExist can be handled as not an error

Reply:
No, because if metafile doesn't exist then it's a problem for all other cases like snapshot delete or snapshot create. In this case use case is different, because even though we were not able to remove a file, we are going to create a fresh file with the same name, returning an error doesn't make any sense as it could be a bad fd and could stuck here indefinitely on next iterations.